smtp/smtp.js

```diff
@@ -51,6 +51,10 @@ class SMTP extends EventEmitter {
 	constructor(options = {}) {
 		super();
 
+		if (options.timeout == null) {
```

Comment:
i guess the way to think about this is that if timeout is not a number, then we default to `TIMEOUT`.

we could try to improve on default values using destructuring: https://wesbos.com/destructuring-default-values/ and then checking for proper types afterwards.

i'll merge this for now, but something to keep in mind.